pyccel/decorators.py

```diff
@@ -139,7 +140,24 @@ class KernelAccessor:
         def __init__(self, f):
             self._f = f
         def __getitem__(self, args):
-            return self._f
+            num_blocks, num_threads = args
+            def internal_loop(*args, **kwargs):
+                """
+                The internal loop for kernel execution.
+
+                The internal loop for kernel execution.
+                """
+                for b in range(num_blocks):
+                    for t in range(num_threads):
+                        cu = CudaThreadIndexing(b, t)
+                        if 'cuda' in self._f.__globals__:
+                            self._f.__globals__['cuda'].threadIdx = cu.threadIdx
+                            self._f.__globals__['cuda'].blockIdx = cu.blockIdx
+                            self._f.__globals__['cuda'].blockDim = cu.blockDim
```

Comment:
Please update the docs to make it clear that the user cannot use syntax such as:
```
from pyccel.decorators import kernel
from pyccel import cuda as py_cu

@kernel
def f():
    print(py_cu.block_idx)
```

Alternatively you may want to search for the correct key by iterating over `self._f.__globals__` and comparing the values to the `pyccel.cuda` module object

Reply:
I think the best thing is to inform the user about the valid and invalid syntax when we have the future in Pyccel. Regarding the documentation, I will add that in the PR when we have thread indexing. Or do you think it's better to add that now ??

Reply:
As long as you don't forget then it's ok to do it later. I am just worried that we will forget.

Pyccel can handle both syntaxes, it is just on the Python side that it is currently unhandled.

If you don't want to do this in this PR please create an issue

Reply:
[654afa3](https://github.com/pyccel/pyccel-cuda/pull/69/commits/654afa3c2938b406e1d54bf4eba26a407cccf297) Does that look good for now ??

Reply:
Yes great. Thanks